src/ol/renderer/canvas/canvastilelayerrenderer.js

```diff
@@ -365,7 +365,7 @@ ol.renderer.canvas.TileLayer.prototype.prepareFrame =
   /** @type {Array.<number>} */
   var zs = Object.keys(tilesToDrawByZ).map(Number);
   zs.sort(ol.array.numberSafeCompareFunction);
-  var opaque = tileSource.getOpaque();
+  var opaque = tileSource.getOpaque(projection);
```

Comment:
Re-reading your original description, it sounds like you are talking about the reprojection resulting in non-opaque pixels being rendered.

Can you provide a screenshot to help illustrate the issue?